src/test/java/org/jenkinsci/plugins/workflow/job/WorkflowRunTest.java

```diff
@@ -491,4 +498,208 @@ private void assertCulprits(WorkflowRun b, String... expectedIds) throws IOExcep
+                checkoutString(sampleRepo, false, false) +
+                "}\n", false));
+
+
+        WorkflowRun b2 = p.scheduleBuild2(0).waitForStart();
+
+        r.assertBuildStatusSuccess(r.waitForCompletion(b2));
+        assertThat(b2.checkouts(listener).size(), equalTo(1));
+        //proving the baseline is removed
+        assertThat(b2.checkouts(listener).get(0).pollingBaseline, nullValue());
+    }
+
+    @Test
+    @Issue("NGPIPELINE-917")
+    // Showing how only with both changelog/poll false is when it removes the checkout
+    public void baselineResetSingleRepoEdgeCases() throws Exception {
+        sampleRepo.init();
+        TaskListener listener = StreamTaskListener.fromStdout();
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        //pipeline definition with both true
+        p.setDefinition(new CpsFlowDefinition(
+                "node {\n" +
+                checkoutString(sampleRepo, true, true) +
+                "}\n", false));
+
+        WorkflowRun b1 = p.scheduleBuild2(0).waitForStart();
+        r.assertBuildStatusSuccess(r.waitForCompletion(b1));
+        assertThat(b1.checkouts(listener).size(), equalTo(1));
+        assertThat(b1.checkouts(listener).get(0).pollingBaseline, notNullValue());
+
+        // Definition when one is false and one is true
+        p.setDefinition(new CpsFlowDefinition(
+                "node {\n" +
+                checkoutString(sampleRepo, false, true) +
+                "}\n", false));
+
+
+        WorkflowRun b2 = p.scheduleBuild2(0).waitForStart();
+
+        r.assertBuildStatusSuccess(r.waitForCompletion(b2));
+        assertThat(b2.checkouts(listener).size(), equalTo(1));
+        assertThat(b2.checkouts(listener).get(0).pollingBaseline, notNullValue());
+
+        // The reverse
+        p.setDefinition(new CpsFlowDefinition(
+                "node {\n" +
+                checkoutString(sampleRepo, true, false) +
+                "}\n", false));
+
+
+        WorkflowRun b3 = p.scheduleBuild2(0).waitForStart();
+
+        r.assertBuildStatusSuccess(r.waitForCompletion(b3));
+        assertThat(b3.checkouts(listener).size(), equalTo(1));
+        assertThat(b3.checkouts(listener).get(0).pollingBaseline, notNullValue());
+
+        // Now it should finally be cleaned up
+        p.setDefinition(new CpsFlowDefinition(
+                "node {\n" +
+                checkoutString(sampleRepo, false, false) +
+                "}\n", false));
+
+
+        WorkflowRun b4 = p.scheduleBuild2(0).waitForStart();
+
+        r.assertBuildStatusSuccess(r.waitForCompletion(b4));
+        assertThat(b4.checkouts(listener).size(), equalTo(1));
+        //Proving it is reset
+        assertThat(b4.checkouts(listener).get(0).pollingBaseline, nullValue());
+    }
+
+    @Test
+    @Issue("NGPIPELINE-917")
+    // Basic test to prove that checkout/poll gets individually set for each checkout
+    public void baselineResetMultipleRepos() throws Exception {
```

Comment:
nit: I feel like between the single-repo tests and `baselineResetMultipleSameRepo` we cover all of the interesting behavior, so I'd just delete this test.